vendor/github.com/prometheus/prometheus/pkg/labels/matcher.go

```diff
@@ -46,8 +46,7 @@ func NewMatcher(t MatchType, n, v string) (*Matcher, error) {
 		Value: v,
 	}
 	if t == MatchRegexp || t == MatchNotRegexp {
-		m.Value = "^(?:" + v + ")$"
-		re, err := regexp.Compile(m.Value)
+		re, err := regexp.Compile("^(?:" + m.Value + ")$")
```

Comment:
This seems to be the cause of the regex information showing up in the ui since I'm using the internal parser to parse silence matchers.

Is this change acceptable? All tests are still passing